docs/DeveloperGuide.md

```diff
@@ -304,12 +343,146 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+**Extensions**
+
+* 1a. There is further input after the user enters list. E.g. `list shows`
+
+    * 1a1. Trackermon will only register the `list` command.
 
       Use case resumes at step 2.
 
+* 2a. The list is empty.
+
+  Use case ends.
+
+---
+**Use case: UC04 - Mark a show**
+
+**Preconditions: Trackermon application is started.**
+
+**Guarantees: Show will be marked in Trackermon only if the user input matches the `mark` format.**
+
+**MSS**
+
+1.  User requests to list shows.
+2.  Trackermon shows a list of shows.
+3.  User requests to mark a specific show in the list.
+4.  Trackermon marks the show.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. Trackermon shows an error message, indicating the format for marking shows is incorrect, and attaches the correct syntax format.
+
+      Use case resumes at step 3.
+---
+**Use case: UC05 - Unmark a show**
+
+**Preconditions: Trackermon application is started.**
+
+**Guarantees: Show will be unmarked in Trackermon only if the user input matches the `unmark` format.**
+
+**MSS**
+
+1.  User requests to list shows.
+2.  Trackermon shows a list of shows.
+3.  User requests to unmark a specific show in the list since the show was not watched yet and marked on accident.
+4.  Trackermon unmarks the show.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. Trackermon shows an error message, indicating the format for unmarking shows is incorrect, and attaches the correct syntax format.
+
+      Use case resumes at step 3.
+
+---
+**Use case: UC06 - Tag a show**
+
+**Preconditions: Trackermon application is started.**
+
+**Guarantees: Show will be tagged in Trackermon only if the user input matches the `add` format.**
+
+**MSS**
+
+1.  User requests to list shows.
+2.  Trackermon shows a list of shows.
+3.  User requests to tag a specific show in the list.
+4.  Trackermon tags the show.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. User uses `add` command with the wrong syntax.
+
+    * 3a1. Trackermon shows an error message to user, indicating the format for adding shows is incorrect, and attaches the correct syntax format.
+
+      Use case resumes at step 3.
+    
+---
+**Use case: UC07 - Exiting Trackermon**
+
+**Preconditions: Trackermon application is started.**
+
+**Guarantees: Trackermon will exit and display an exit message for the user.**
+
+**MSS**
+
+1. User enters `exit` command in Trackermon.
+2. Trackermon saves all changes into the disk.
+3. Trackermon displays a message to tell the user that the app has successfully closed.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. There is further input after the user enters exit. E.g. `exit now`
+
+    * 1a1. Trackermon will only register the `exit` command.
+
+      Use case resumes at step 2.
+---  
```

Comment:
Are we implementing the exit command for v1.2? Just checking to see if we might want to add that to our feature list if it seems needed.

Reply:
Seconded. I think that an exit command seems like a basic feature for a minimal working product. We can confirm with the rest during the next team meeting. In this case, we should also consider adding it into the user guide as while.